autogen/src/main.rs

```diff
@@ -76,16 +167,459 @@ async fn update_json() -> Result<()> {
+    for val in enm {
+      let val: String = format!("{val}").chars().filter(|ch| ch != &'"').collect();
+      writeln!(writer, "{val}")?;
+    }
+    writer.unindent();
+    writeln!(writer, "}}")?;
+  } else if let Some(list) = typ.subschemas.as_ref().and_then(|ss| ss.any_of.as_ref()) {
+    if list.is_empty() {
+      scalars.push(name);
+      return Ok(());
+    }
+    writeln!(writer, "input {name} {{")?;
+    writer.indent();
+    for property in list {
+      let property = property.clone().into_object();
+      let description = property
+        .metadata
+        .as_ref()
+        .and_then(|metadata| metadata.description.as_ref());
+      write_description(writer, description)?;
+      if let Some(obj) = property.object {
+        for (name, schema) in obj.properties {
+          write_field(writer, name, schema.into_object(), defs, extra_it)?;
+        }
+      }
+    }
+    writer.unindent();
+    writeln!(writer, "}}")?;
+  } else if let Some(list) = typ.subschemas.as_ref().and_then(|ss| ss.one_of.as_ref()) {
+    if list.is_empty() {
+      scalars.push(name);
+      return Ok(());
+    }
+    writeln!(writer, "input {name} {{")?;
+    writer.indent();
+    for property in list {
+      if let Some(obj) = property.clone().into_object().object {
+        for (name, schema) in obj.properties {
+          write_field(writer, name, schema.into_object(), defs, extra_it)?;
+        }
+      }
+    }
+    writer.unindent();
+    writeln!(writer, "}}")?;
+  } else if let Some(SingleOrVec::Single(item)) = typ.array.and_then(|arr| arr.items) {
+    if let Some(name) = item.into_object().reference {
+      writeln!(writer, "{name}")?;
+    } else {
+      scalars.push(name);
+    }
+  }
+
+  Ok(())
+}
+
+fn write_property(
+  writer: &mut IndentedWriter<impl Write>,
+  name: String,
+  property: Schema,
+  defs: &BTreeMap<String, Schema>,
+  extra_it: &mut BTreeMap<String, ObjectValidation>,
+) -> std::io::Result<()> {
+  // println!("Property: name = {name}");
+  let property = property.into_object();
+  let description = property
+    .metadata
+    .as_ref()
+    .and_then(|metadata| metadata.description.as_ref());
+  write_description(writer, description)?;
+  write_field(writer, name, property, defs, extra_it)?;
+  Ok(())
+}
+
+fn write_schema(
+  writer: &mut IndentedWriter<impl Write>,
+  mut name: String,
+  schema: SchemaObject,
+  defs: &BTreeMap<String, Schema>,
+  on: &str,
+  written_directives: &mut HashSet<String>,
+  extra_it: &mut BTreeMap<String, ObjectValidation>,
+) -> std::io::Result<()> {
+  // if written_directives.contains(&name) {
+  //   return Ok(());
+  // }
+  // println!("{name}: {:?}", ());
+  let description = schema
+    .metadata
+    .as_ref()
+    .and_then(|metadata| metadata.description.as_ref());
+  write_description(writer, description)?;
+  unsafe {
+    name.as_bytes_mut().get_mut(0).map(|ch| {
+      let lower = (*ch as char).to_ascii_lowercase();
+      *ch = lower as u8;
+    });
+  }
```

Comment:
maybe just use some crate instead of playing with unsafe?

Reply:
removed unsafe